src/index.tsx

```diff
@@ -1,54 +1,28 @@
-import {
-  addons,
-  makeDecorator,
-  WrapperSettings,
-  StoryWrapper,
-} from '@storybook/addons';
+import { StoryWrapper, useMemo } from '@storybook/addons';
 
-import { DEFAULT_PADDING, EVENTS, PARAM_KEY } from './constants';
-import { getSelectedPadding, isEnabled } from './helpers';
-
-const state: {
-  initialized: boolean;
-  padding?: string;
-} = {
-  initialized: false,
-};
+import { DEFAULT_PADDING, PARAM_KEY } from './constants';
+import { normalizeValues, getSelectedPadding } from './helpers';
 
 const setStyle = (padding = DEFAULT_PADDING) => {
   document.body.style.padding = padding;
   document.body.style.transition = 'padding .3s';
 };
 
-const setPadding = (padding: string) => {
-  state.padding = padding;
-  setStyle(padding);
-};
-
-const render = (
-  settings: WrapperSettings,
-  storyFn: () => ReturnType<StoryWrapper>,
-) => {
-  const { parameters: options = {} } = settings;
-  const channel = addons.getChannel();
-
-  if (!state.initialized) {
-    channel.on(EVENTS.UPDATE, setPadding);
-    state.initialized = true;
-  }
-
-  const currentValue = isEnabled(options) ? state.padding : DEFAULT_PADDING;
-  setStyle(getSelectedPadding(options, currentValue));
-
-  return storyFn();
+export const withPaddings: StoryWrapper = (getStory, context) => {
+  const { globals, parameters } = context;
+  const globalsSelectedPadding = globals[PARAM_KEY]?.value;
+  const paddingsConfig = parameters[PARAM_KEY];
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
Let me take a look on this one, I think this was mostly related to the fact that the name of the component is not named as `use` something. (I think we discussed it briefly somewhere). 

Let me double check a bit and see if I can do anything about it, but i don't think it's related to the deps.

Reply:
hey @rbardini, I think this one is the last item, basically this complains as mentioned above, because of the name pattern. Since linting rule it's treating it as a regular component it looks for the Capital case name. What do you think if instead using the `withPaddings` naming, we rename it to `usePaddings`? This way it's compliant with the rule and at the end it's a hook anyway, so it would make sense as well.

Reply:
I checked the backgrounds addon and it follows the same naming convention:

https://github.com/storybookjs/storybook/blob/c8abe86cde37e70d98ef2f5ab3051dbee9df0bf8/addons/backgrounds/src/decorators/withBackground.ts#L6

so let's keep the disable comment for now 🙂